extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/NarayanaJtaRecorder.java

```diff
@@ -32,34 +31,115 @@ public void setNodeName(final TransactionManagerConfiguration transactions) {
         }
     }
 
-    public void setDefaultProperties(Properties properties) {
-        //TODO: this is a huge hack to avoid loading XML parsers
-        //this needs a proper SPI
-        try {
-            Field field = PropertiesFactory.class.getDeclaredField("delegatePropertiesFactory");
-            field.setAccessible(true);
-            field.set(null, new QuarkusPropertiesFactory(properties));
-
-        } catch (Exception e) {
-            log.error("Could not override transaction properties factory", e);
-        }
-
-        defaultProperties = properties;
-    }
-
     public void setDefaultTimeout(TransactionManagerConfiguration transactions) {
         transactions.defaultTransactionTimeout.ifPresent(defaultTimeout -> {
             arjPropertyManager.getCoordinatorEnvironmentBean().setDefaultTimeout((int) defaultTimeout.getSeconds());
             TxControl.setDefaultTimeout((int) defaultTimeout.getSeconds());
         });
     }
 
-    public static Properties getDefaultProperties() {
-        return defaultProperties;
-    }
-
     public void disableTransactionStatusManager() {
         arjPropertyManager.getCoordinatorEnvironmentBean()
                 .setTransactionStatusManagerEnable(false);
     }
+
+    private void setXaResourceOrphanFilterClassNames(JTAEnvironmentBean jtaEnvironmentBean) {
+        jtaEnvironmentBean.setXaResourceOrphanFilterClassNames(Arrays.asList(
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.JTATransactionLogXAResourceOrphanFilter",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.JTANodeNameXAResourceOrphanFilter",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinateJTAXAResourceOrphanFilter",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinationManagerXAResourceOrphanFilter"));
+    }
+
+    private void setRecoveryModuleClassNames(RecoveryEnvironmentBean recoveryEnvironmentBean) {
+        recoveryEnvironmentBean.setRecoveryModuleClassNames(Arrays.asList(
+                "com.arjuna.ats.internal.arjuna.recovery.AtomicActionRecoveryModule",
+                "com.arjuna.ats.internal.txoj.recovery.TORecoveryModule",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinateAtomicActionRecoveryModule",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule",
+                "com.arjuna.ats.internal.jta.recovery.arjunacore.CommitMarkableResourceRecordRecoveryModule"));
+    }
+
+    private void setExpiryScannerClassNames(RecoveryEnvironmentBean recoveryEnvironmentBean) {
+        recoveryEnvironmentBean.setExpiryScannerClassNames(Collections.singletonList(
+                "com.arjuna.ats.internal.arjuna.recovery.ExpiredTransactionStatusManagerScanner"));
+    }
+
+    private void setRecoveryPort(RecoveryEnvironmentBean recoveryEnvironmentBean) {
+        recoveryEnvironmentBean.setRecoveryPort(4712);
```

Comment:
would not be good to make this configurable?

Reply:
It is configurable via a system property.

Reply:
But only if you decide to use the system properties "configuration model", right?

Reply:
Exactly. It is a conscious decision to be made by the app developer.